### PR TITLE
Protect MinimizerOptions::DefaultMinimizerType from multi-threaded calls

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

File: math/mathcore/src/MinimizerOptions.cxx [hunk 1]
```diff
@@ -92,15 +94,25 @@ const std::string & MinimizerOptions::DefaultMinimizerType()
...
+      return Minim::gDefaultMinimizer;
+
+   R__WRITE_LOCKGUARD(ROOT::gCoreMutex);
+
+   // Another thread also waiting for the write lock might have
+   // done the assignment
+   if (Minim::gDefaultMinimizer.size() != 0)
+      return Minim::gDefaultMinimizer;
+
 #ifndef MATH_NO_PLUGIN_MANAGER
```
Comment (dpiparo) [inline math/mathcore/src/MinimizerOptions.cxx]: Do we know why we have this lockguard?

**Comment by lmoneta** [comment]
I think we can merge this PR. If you want I can do it

**Comment by Dr15Jones** [comment]
@pcanal can this be backported to 6.12?

**Comment by pcanal** [comment]
Done:
* 646e20138a - (6 hours ago) Respect the semantic of MATH_NO_PLUGIN_MANAGER in DefaultMinimizerType — Philippe Canal (HEAD -> v6-12-00-patches, origin/v6-12-00-patches)
* bd8476c068 - (18 hours ago) Protect MinimizerOptions::DefaultMinimizerType from multi-threaded calls — Philippe Canal

but too late for 6.12/06

**Pull request merged**